src/sfeQwiicUltrasonic.cpp

```diff
@@ -29,13 +40,33 @@ sfeTkError_t sfeQwiicUltrasonic::isConnected()
     return _theBus->ping();
 }
 
-sfeTkError_t sfeQwiicUltrasonic::triggerAndRead(uint16_t &distance)
+sfeTkError_t sfeQwiicUltrasonic::getDistance(uint16_t &distance)
+{
+    size_t bytesRead = 0;
+    uint8_t rawData[2] = {0, 0};
+    sfeTkError_t err;
+
+    _theBus->writeByte(kUltrasonicDistanceReadCommand);
+    err = _theBus->readRegisterRegion(_theBus->address(), rawData, 2, bytesRead);
+
+    // Check whether the read was successful
+    if (err != kSTkErrOk)
+        return err;
+
+    // Store raw data
+    distance = (rawData[0] << 8) | rawData[1];
+
+    // Done!
+    return kSTkErrOk;
+}
+
+sfeTkError_t sfeQwiicUltrasonic::getTriggeredDistance(uint16_t &distance)
 {
     size_t bytesRead = 0;
     uint8_t rawData[2] = {0, 0};
 
     // Attempt to read the distance
-    sfeTkError_t err = _theBus->readRegisterRegion(kQwiicUltrasonicRegisterTrigger, rawData, 2, bytesRead);
+    sfeTkError_t err = _theBus->readRegisterRegion(_theBus->address(), rawData, 2, bytesRead);
```

Comment:
This doesn't make sense to me? The first parameter is the register address, not the device address. Have you tested that this actually works?

Reply:
Yeah I did and couldn't believe that it did. I was being a bit hasty to be honest. I have now added a few more methods to the toolkit to make this make sene.